sdks/java/extensions/zetasketch/src/main/java/org/apache/beam/sdk/extensions/zetasketch/HllCount.java

```diff
@@ -0,0 +1,350 @@
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.zetasketch;
+
+import com.google.zetasketch.HyperLogLogPlusPlus;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+
+/**
+ * {@code PTransform}s to compute HyperLogLogPlusPlus (HLL++) sketches on data streams based on the
+ * <a href="https://github.com/google/zetasketch">ZetaSketch</a> implementation.
+ *
+ * <p>HLL++ is an algorithm implemented by Google that estimates the count of distinct elements in a
+ * data stream. HLL++ requires significantly less memory than the linear memory needed for exact
+ * computation, at the cost of a small error. Cardinalities of arbitrary breakdowns can be computed
+ * using the HLL++ sketch. See this <a
+ * href="http://static.googleusercontent.com/media/research.google.com/en/us/pubs/archive/40671.pdf">published
+ * paper</a> for details about the algorithm.
+ *
+ * <p>HLL++ functions are also supported in <a
+ * href="https://cloud.google.com/bigquery/docs/reference/standard-sql/hll_functions">Google Cloud
+ * BigQuery</a>. Using the {@code HllCount PTransform}s makes the interoperation with BigQuery
+ * easier.
+ *
+ * <p>For detailed design of this class, see https://s.apache.org/hll-in-beam.
+ *
+ * <h3>Examples</h3>
+ *
+ * <h4>Example 1: Create long-type sketch for a {@code PCollection<Long>} and specify precision</h4>
+ *
+ * <pre>{@code
+ * PCollection<Long> input = ...;
+ * int p = ...;
+ * PCollection<byte[]> sketch = input.apply(HllCount.Init.longSketch().withPrecision(p).globally());
+ * }</pre>
+ *
+ * <h4>Example 2: Create bytes-type sketch for a {@code PCollection<KV<String, byte[]>>}</h4>
+ *
+ * <pre>{@code
+ * PCollection<KV<String, byte[]>> input = ...;
+ * PCollection<KV<String, byte[]>> sketch = input.apply(HllCount.Init.bytesSketch().perKey());
+ * }</pre>
+ *
+ * <h4>Example 3: Merge existing sketches in a {@code PCollection<byte[]>} into a new one</h4>
+ *
+ * <pre>{@code
+ * PCollection<byte[]> sketches = ...;
+ * PCollection<byte[]> mergedSketch = sketches.apply(HllCount.MergePartial.globally());
+ * }</pre>
+ *
+ * <h4>Example 4: Estimates the count of distinct elements in a {@code PCollection<String>}</h4>
+ *
+ * <pre>{@code
+ * PCollection<String> input = ...;
+ * PCollection<Long> countDistinct =
+ *     input.apply(HllCount.Init.stringSketch().globally()).apply(HllCount.Extract.globally());
+ * }</pre>
+ *
+ * Note: Currently HllCount does not work on FnAPI workers. See <a
+ * href="https://issues.apache.org/jira/browse/BEAM-7879">Jira ticket [BEAM-7879]</a>.
+ */
+@Experimental
+public final class HllCount {
+
+  public static final int MINIMUM_PRECISION = HyperLogLogPlusPlus.MINIMUM_PRECISION;
```

Comment:
If I've seen this correctly, all classes which use these constants import HyperLogLogPlusPlus anyways... so the re-direct here seems to have limited benefits. How about just using the HyperLogLogPlusPlus constants directly? Or what was your rationale?

Reply:
Oh, I added them here not for re-directing for the other helper classes. This is because I want these numbers to be visible to users, because they need to know the numbers to set a proper precision. These constants are also referred to from the Javadoc comments that users can see and click.

Reply:
Ah, got it! But the Javadoc does not expand the actual numbers, or does it? 
Maybe leave a comment with "constants replicated here for documentation purposes / to make them easy to find for users"?

Reply:
Ah, thanks for catching this. Now I have set up javadoc correctly such that the actual numbers can be seen, and I think it is clear now that those are for users.